### PR TITLE
Add the ability to select the debugger engines to connect to

### PR DESCRIPTION
**Pull request by yitzhaks**
This includes a new dialog which presents the available engines, similar to the Visual Studio dialog. The default is still to use the default engine.

### CODE REVIEW

File: src/WacVsTools.Core/AttachToWacProcess/AttachToWacProcessDialog.xaml.cs [hunk 1]
```diff
@@ -1,20 +1,24 @@
...
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Windows;
+    using System.Windows.Controls;
+
     public partial class AttachToWacProcessDialog : Window
     {
-        List<int> selectedProcesses;
+        private AttachToWacProcessDialogModel m_model;
+        private HashSet<int> m_selectedProcesses;
```
Comment (MarcinJuraszek) [inline src/WacVsTools.Core/AttachToWacProcess/AttachToWacProcessDialog.xaml.cs]: please don't use `m_` when naming fields.

File: src/WacVsTools.Core/AttachToWacProcess/SelectDebuggerEngineDialogModel.cs [hunk 2]
```diff
@@ -0,0 +1,248 @@
...
+                m_isLazy,
+                m_isAutomatic,
+                m_manualSelection
+                    .Select(debuggerEngine => debuggerEngine.Clone()).ToList());
+        }
+    }
+
+    public sealed class DebuggerEngine : INotifyPropertyChanged, IComparable<DebuggerEngine>
+    {
+        private bool m_isSelected;
```
Comment (MarcinJuraszek) [inline src/WacVsTools.Core/AttachToWacProcess/SelectDebuggerEngineDialogModel.cs]: nit: `m_`

File: src/WacVsTools.Core/AttachToWacProcess/SelectDebuggerEngineDialogModel.cs [hunk 3]
```diff
@@ -0,0 +1,248 @@
...
+        private const string RegistryKeyPath = @"SOFTWARE\Microsoft\WAC\WacVsTools";
+        private const string IsAutomaticValueName = @"IsAutomatic";
+        private const string ManuallySelectedEnginesValueName = @"ManuallySelectedEngines";
+
+        public const string DefaultDebuggerEngineName = @"Managed (v4.6, v4.5, v4.0) code";
+        public const string DefaultDebuggerEngineID = @"{FB0D4648-F776-4980-95F8-BB7F36EBC1EE}";
+
+        private static readonly DebuggerEngines s_default = new DebuggerEngines(isLazy: true, isAutomatic: true, manualSelection: new[] { new DebuggerEngine(DefaultDebuggerEngineName, DefaultDebuggerEngineID, isSelected: false) });
+
+        private readonly bool m_isLazy;
```
Comment (MarcinJuraszek) [inline src/WacVsTools.Core/AttachToWacProcess/SelectDebuggerEngineDialogModel.cs]: `m_`

**Review by MarcinJuraszek — CHANGES_REQUESTED** [review]

File: src/WacVsTools.Core/AttachToWacProcess/AttachToWacProcessDialog.xaml.cs (hunk 1 above)
Comment (yitzhaks) [reply src/WacVsTools.Core/AttachToWacProcess/AttachToWacProcessDialog.xaml.cs]: :(

**Review by MarcinJuraszek — APPROVED** [review]

**Pull request merged**